### ISSUE TITLE
Enable code coverage using latest version of lcov (1.14)

### ISSUE THREAD

**Issue by ghost** [i3/i3#4161]
## I'm submitting a…

<pre>
[ ] Bug
[x] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

After cloning the git repository and running autoreconf, I am unable to run configure with enabled code coverage. 

<pre>
`../configure --enable-code-coverage

checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking for gsed... sed
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking whether to enable maintainer-specific portions of Makefiles... yes
checking for gawk... (cached) gawk
checking for GNU make... make
checking for a sed that does not truncate output... /usr/bin/sed
checking whether to build with code coverage support... yes
checking for gcov... gcov
checking for lcov... lcov
checking for genhtml... genhtml
checking for lcov version... invalid
configure: error: You must have one of the following versions of lcov: 1.6 1.7 1.8 1.9 1.10 1.11 1.12 1.13 (found:
1.14).
`</pre>

## Desired Behavior

Code coverage is generated using lcov 1.14

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
i3 version:  4.18.2 (2020-07-26)
</pre>

<pre>
- Linux Distribution & Version: Arch Linux
- Are you using a compositor (e.g., xcompmgr or compton): xcompmgr
</pre>

**Comment by stapelberg** [comment]
Does it work if you’re using meson? We’re switching away from autotools, see #4086

**Comment by wlhlm** [comment]
lcov 1.14 works with meson. There is another issue however: When modifying the i3 codebase after running `meson test` (with coverage enabled), subsequent test runs will fail with errors like:
```
#   Failed test 'single number (move workspace 3) ok'
#   at /home/wlhlm/Devel/i3/testcases/t/187-commands-parser.t line 46.
#          got: 'cmd_move_con_to_workspace_name(3, (null))
# libgcov profiling error:/home/wlhlm/Devel/i3/build/test.commands_parser.p/src_commands_parser.c.gcda:overwriting an existing profile data with a different timestamp'
#     expected: 'cmd_move_con_to_workspace_name(3, (null))'
```
My workaround has been to run `find . -name \*.gcno -or -name \*.gcda -delete` in the build dir.

**Comment by stapelberg** [comment]
Hm! I wonder if that’s an lcov or meson issue. I doubt we’re doing anything wrong in i3, as code coverage is entirely handled by meson AFAIK.

**Comment by wlhlm** [comment]
This happens with autotools as well.

**Comment by stapelberg** [comment]
That implies it’s an lcov issue then. Could you report it there?

**Comment by wlhlm** [comment]
TBH, this is the first time I'm working with anything lcov/gcov-related, not sure if I can make a helpful bug report (and motivation is pretty low).

Searching around, while not finding a lot of related stuff, it looks like this has been introduced intentionally on the gcov-side: <https://gcc.gnu.org/legacy-ml/gcc-patches/2018-05/msg01608.html> ([follow-up](https://www.mail-archive.com/gcc-patches@gcc.gnu.org/msg205422.html) modifying the error message slightly). So maybe we are doing something wrong on our side?

**Comment by stapelberg** [comment]
We have released 4.19 recently which switched to meson.

Any remaining issues should be reported with the meson or gcov projects, until there is evidence that i3 is doing something wrong.

Thanks,
